<launch>
  <arg name="sd_vehicle" default="env200" doc="the vehicle under control, env200 or twizy"/>
  <arg name="sd_gps_imu" default="oxts" doc="The Gps or IMU, oxts or peak"/>
  <arg name="sd_speed_source" default="vehicle_can_speed" doc="The source of vehicle speed used to close the control loop, vehicle_can_speed, imu_speed or ndt_speed"/>
  <arg name="sd_simulation_mode" default="false" doc="Choose to launch the VI for control of a real vehicle or simulated vehicle, true for simulation (e.g Gazebo)"/>

  <node pkg="sd_vehicle_interface" type="sd_vehicle_interface_node" name="sd_vehicle_interface_node" output = "screen">
    	<param name="sd_vehicle" type="string" value="$(arg sd_vehicle)" />
    	<param name="sd_gps_imu" type="string" value="$(arg sd_gps_imu)" />
    	<param name="sd_speed_source" type="string" value="$(arg sd_speed_source)" />
    	<param name="sd_simulation_mode" type="bool" value="$(arg sd_simulation_mode)" />
	</node>

<group unless="$(arg sd_simulation_mode)">
  <node pkg="sd_vehicle_interface" type="socketcan_bridge_node" name="socketcan_bridge_node" output = "screen"/>
</group>


</launch>
